lib/po_libc_wrappers.c

```diff
@@ -132,8 +137,9 @@ stat(const char *path, struct stat *st)
 struct po_map*
 po_map_get()
 {
+	//If a global map is not found, do not create it, instead attempt to look in the shared memory.
 	if (global_map == NULL) {
-		global_map = po_map_create(4);
```

Comment:
In fact, this code can probably be simplified even further. The `static struct po_map *global_map` can only be accessed by code in this translation unit, which never leaks the map nor causes the refcount to be decremented. So, we don't actually need this refcounting behaviour. Instead, you can probably fold the `po_map_get()` functionality directly into `get_shared_map()`, a la:

```c
if (global_map) {
    po_map_assertvalid(global_map);
    return (global_map);
}
```

Similarly, instead of calling `po_map_set()`, you can just assign to `global_map`. Then you can remove the `po_map_{get,set}()` functions and have easier-to-understand code.